cpp/src/arrow/flight/sql/odbc/tests/errors_test.cc

```diff
@@ -304,10 +307,10 @@ TYPED_TEST(ErrorsTest, TestSQLGetDiagRecInputData) {
   EXPECT_EQ(SQL_NO_DATA, SQLGetDiagRec(SQL_HANDLE_DBC, this->conn, 1, 0, 0, 0, 0, 0));
 
   // Invalid handle
-  EXPECT_EQ(SQL_INVALID_HANDLE, SQLGetDiagRec(0, 0, 0, 0, 0, 0, 0, 0));
+  EXPECT_EQ(SQL_INVALID_HANDLE, SQLGetDiagRec(0, this->conn, 1, 0, 0, 0, 0, 0));
 }
```

Comment:
we still want to test passing all 0's on Windows where it works. Here is my pseudo-code suggestion: 
```
ifdef win32
EXPECT_EQ(SQL_INVALID_HANDLE, SQLGetDiagRec(0, 0, 0, 0, 0, 0, 0, 0));
else if __APPLE__
// macOS ODBC driver manager requires connection handle 
EXPECT_EQ(SQL_INVALID_HANDLE, SQLGetDiagRec(0, this->conn, 1, 0, 0, 0, 0, 0));
endif 
```